src/bin/bindgen.rs

```diff
@@ -217,15 +217,26 @@ pub fn main() {
 
     let mut bind_args: Vec<_> = env::args().collect();
 
-    let clang_version = clang_version();
-
-    if cfg!(feature = "llvm_stable") {
-        assert_eq!(clang_version, "3.8.0");
-    } else {
-        assert_eq!(clang_version, "3.9.0");
-    }
-
-    debug!("Using Clang Version: {}", clang_version);
+    match clang_version() {
```

Comment:
With the other suggested changes we'd do something like:

```rust
let version = clang_version();
let expected_version = if cfg!(feature = "llvm_stable") { (3, 8) } else { (3, 9) };

info!("Using clang {}", version.full);

match version.parsed { // Or however you call the member
    None => warn!("Couldn't parse libclang version"),
    Some(version) if version != expected_version => {
        error!("Using clang {:?}, expected {:?}", version, expected_version);
    }
    _ => {}
}
```

Which I think reads a bit better and duplicates less code :)